test/e2e/framework/skipper/skipper.go

```diff
@@ -56,50 +55,6 @@ func SkipUnlessAtLeast(value int, minValue int, message string) {
 	}
 }
 
-var featureGate featuregate.FeatureGate
-
-// InitFeatureGates must be called in test suites that have a --feature-gates parameter.
-// If not called, SkipUnlessFeatureGateEnabled and SkipIfFeatureGateEnabled will
-// record a test failure.
-func InitFeatureGates(defaults featuregate.FeatureGate, overrides map[string]bool) error {
-	clone := defaults.DeepCopy()
-	if err := clone.SetFromMap(overrides); err != nil {
-		return err
-	}
-	featureGate = clone
-	return nil
-}
-
-// SkipUnlessFeatureGateEnabled skips if the feature is disabled.
-//
-// Beware that this only works in test suites that have a --feature-gate
-// parameter and call InitFeatureGates. In test/e2e, the `Feature: XYZ` tag
-// has to be used instead and invocations have to make sure that they
-// only run tests that work with the given test cluster.
-func SkipUnlessFeatureGateEnabled(gate featuregate.Feature) {
-	if featureGate == nil {
-		framework.Failf("Feature gate checking is not enabled, don't use SkipUnlessFeatureGateEnabled(%v). Instead use the Feature tag.", gate)
-	}
-	if !featureGate.Enabled(gate) {
-		skipInternalf(1, "Only supported when %v feature is enabled", gate)
-	}
-}
-
-// SkipIfFeatureGateEnabled skips if the feature is enabled.
-//
-// Beware that this only works in test suites that have a --feature-gate
-// parameter and call InitFeatureGates. In test/e2e, the `Feature: XYZ` tag
-// has to be used instead and invocations have to make sure that they
-// only run tests that work with the given test cluster.
-func SkipIfFeatureGateEnabled(gate featuregate.Feature) {
-	if featureGate == nil {
-		framework.Failf("Feature gate checking is not enabled, don't use SkipFeatureGateEnabled(%v). Instead use the Feature tag.", gate)
-	}
-	if featureGate.Enabled(gate) {
-		skipInternalf(1, "Only supported when %v feature is disabled", gate)
-	}
-}
-
```

Comment:
If `e2e_node` was the only potential user of this code and even that doesn't really use the "skipping" aspect of it, then I agree that this code should get removed.

My last cleanup was an attempt to catch incorrect usage of `SkipUnlessFeatureGateEnabled/SkipIfFeatureGateEnabled`, but that obviously didn't help when the code wasn't even getting called at all.

Reply:
:)
Yeah I was pretty surprised that these functions aren't being called, perhaps they were in the past.

As a side note: IMHO the best way of skipping tests is from the outside rather than in-test skips, since it's sometimes easy to miss a test being skipped.

Reply:
They were used incorrectly in the past, which I fixed at least in some places: 12990dec4057847bf6e7e8037c27baa16e58648b

Reply:
I think all tests needed it GA-ed. We will have a need for it again at some point. I am fine with removing it for now.

Long term the reliable way to test feature gates may be to query /config endpoint of the kubelet.

Reply:
this was a source of confusion, if we can remove it the better, long term we should build something better documented and consistent

Reply:
I'm using `SkipUnlessFeatureGateEnabled` in the test case I'm developing now. It tests CDI injections in the e2e_node/device_plugin_test.go. The test case should be skipped if the future gate `DevicePluginCDIDevices` is not enabled.

What should I use instead?